core/src/main/java/org/infinispan/interceptors/distribution/DistributionBulkInterceptor.java

```diff
@@ -77,8 +80,8 @@ private BackingEntrySet(Cache cache, CacheSet<CacheEntry<K, V>> entrySet, LocalF
 
       @Override
       public CloseableIterator<CacheEntry<K, V>> iterator() {
-         return new CloseableIteratorMapper<>(Closeables.iterator(stream()),
-                 e -> new EntryWrapper<>(cache, e));
+         return new CloseableIteratorMapper<>(new RemovableCloseableIterator<>(Closeables.iterator(stream()), cache,
+                 e -> e.getKey()), e -> new EntryWrapper<>(cache, e));
```

Comment:
nitpick: ```e -> e.getKey()``` could be replaced by ```CacheEntry::getKey```. the same for the change below.
ps. not sure which one is more efficient.